src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -4299,6 +4299,34 @@ bool GraphBuilder::try_inline_full(ciMethod* callee, bool holder_known, bool ign
   caller_state->truncate_stack(args_base);
   assert(callee_state->stack_size() == 0, "callee stack must be empty");
 
+  // Check if we need a membar at the beginning of the java.lang.Object
+  // constructor to satisfy the memory model for strict fields.
+  if (EnableValhalla && method()->intrinsic_id() == vmIntrinsics::_Object_init) {
```

Comment:
```suggestion
  if (EnableValhalla && callee->intrinsic_id() == vmIntrinsics::_Object_init) {
```
Fits better in the context of this function.

Reply:
I would rather leave the code as is because we enter the callee scope just above. Also, code further below also uses `method()`.